### PR TITLE
Add draft of pyproject.toml

### PR DESCRIPTION
**Pull request by uekerman**

### CODE REVIEW

**Review by fsimonis — APPROVED** [review]
Looks correct. I suggest to merge this now and add release-specific things in a followup PR.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+description="A tool to co-simulate FMU models with the coupling library preCICE."
+readme = "README.md"
+license={ text = "LGPL-3.0" }
+keywords = [ "preCICE", "FMI", "ODE", "controller", "PDE", "coupling" ]
+classifiers=[
+"Development Status :: 4 - Beta",
+"License :: OSI Approved :: GNU LESSER General Public License v3 (LGPLv3)",
+"Programming Language :: Python :: 3 :: Only",
+"Programming Language :: Python :: 3.8",
+"Topic :: Scientific/Engineering :: TODO",
```
Comment (fsimonis) [inline pyproject.toml:27]: TODO?

Comment (fsimonis) [reply pyproject.toml:27]: Potential ones are:
* Topic :: Scientific/Engineering
* Topic :: Scientific/Engineering :: Interface Engine/Protocol Translator
* Topic :: Scientific/Engineering :: Mathematics
* Topic :: Scientific/Engineering :: Physics

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+[project]
+name="fmiprecice"
+dynamic = [ "version" ]
+dependencies = [
+    "fmpy", "numpy", "pyprecice>=3.0"
+]
+requires-python = ">=3.8"
+authors = [
+    { name = "The preCICE Developers",  email="info@precice.org"}
```
Comment (fsimonis) [inline pyproject.toml:13]: Do we want to list Leonard Willeke explicitly as an author?

File: pyproject.toml (hunk 1 above)
Comment (uekerman) [reply pyproject.toml:27]: Picked the first one.

File: pyproject.toml (hunk 2 above)
Comment (uekerman) [reply pyproject.toml:13]: I see this mainly as what we want moving forward. I think it's fine like this. We have citation guidelines in the documentation.

**Pull request merged**